grammars/mlir.json

```diff
@@ -1,112 +1,590 @@
+          "name": "comment.line.double-slash.mlir"
+        },
+        "2": {
+          "name": "comment.other.CHECK.mlir"
+        },
+        "3": {
+          "patterns": [
+            {
+              "include": "#CHECK-CODE"
+            }
+          ]
+        },
+        "name": "comment.other.CHECK.mlir"
+      }
+    },
+    "CHECK-CODE": {
+      "match": "([^0-9a-zA-Z%\"@]*)?([0-9a-zA-Z\"\\.%_\\-@]*)([^0-9a-zA-Z\"\\.%_\\-].*)?$",
+      "captures": {
+        "1": {
+          "name": "comment.line.double-slash.mlir"
+        },
+        "2": {
+          "patterns": [
+            {
+              "include": "$self"
+            },
+            {
+              "match": "(\\b|x)(index|i[1-9][0-9]*|f16|bf16|f32|f64|u8|memref|tensor|vector|func)\\b",
+              "captures": {
+                "1": {
+                  "name": "comment.line.double-slash.mlir"
+                },
+                "2": {
+                  "name": "storage.type.mlir"
+                }
+              }
+            },
+            {
+              "match": "\\b([0-9]+)(x.*)?",
+              "captures": {
+                "1": {
+                  "name": "constant.numeric.mlir"
+                },
+                "2": {
+                  "patterns": [
+                    {
+                      "include": "$self"
+                    }
+                  ]
+                }
+              }
+            },
+            {
+              "match": "(.*)",
+              "name": "comment.line.double-slash.mlir"
+            }
+          ]
+        },
+        "3": {
+          "patterns": [
+            {
+              "include": "#CHECK-CODE"
+            }
+          ]
+        }
+      }
     }
   },
-  "patterns":[
+  "patterns": [
     {
-      "include":"#comment"
+      "match": "\\b(func)\\b\\s*(@[\\w_][\\w\\d_.$]*)",
+      "captures": {
+        "1": {
+          "name": "keyword.function.mlir"
+        },
+        "2": {
+          "name": "entity.name.function.mlir"
+        }
+      },
+      "name": "support.function.mlir"
     },
     {
-      "include":"#string"
+      "match": "\\b(attributes|br|call|constant|loc|return)\\b",
+      "name": "keyword.module.mlir"
     },
     {
-      "match":"\\b(func)\\b\\s*(@[\\w_][\\w\\d_.$]*)",
-      "captures":{
-        "1":{
-          "name":"keyword.function.mlir"
-        },
-        "2":{
-          "name":"entity.name.function.mlir"
-        }
-      },
-      "name":"support.function.mlir"
+      "include": "#identifier"
+    },
+    {
+      "include": "#branch_target"
+    },
+    {
+      "include": "#attribute"
+    },
+    {
+      "include": "#memref-size"
+    },
+    {
+      "include": "#numbers"
+    },
+    {
+      "include": "#affineOps"
+    },
+    {
+      "include": "#affineOpsP"
+    },
+    {
+      "include": "#affineStructures"
+    },
+    {
+      "include": "#else"
+    },
+    {
+      "include": "#CHECK"
+    },
+    {
+      "include": "#llvmType"
```

Comment:
Could we makes these into categories? E.g., include "#llvmDialect" here and then have llvmDialect include all of these to avoid a flat list here.